tests/test_frompyobject.rs

```diff
@@ -365,17 +365,17 @@ fn test_enum() {
             _ => panic!("Expected extracting Foo::StructVarGetAttrArg, got {:?}", f),
         }
 
-        let dict = PyDict::new(py);
+        let dict = PyDict::new_bound(py);
         dict.set_item("a", "test").expect("Failed to set item");
-        let f = Foo::extract(dict.as_ref()).expect("Failed to extract Foo from dict");
+        let f = Foo::extract_bound(dict.as_ref()).expect("Failed to extract Foo from dict");
```

Comment:
I think generally I prefer the turbofish style here, as it'll be less residual work when we tidy up `FromPyObject` in the future:

```suggestion
        let f = dict.extract::<Foo>().expect("Failed to extract Foo from dict");
```

and similar for a lot of other cases in this file.

Reply:
I've updated the ones related to my changes here (or would you like to also include the ones not relevant to the `PyDict` conversion?)

Reply:
Ah, let's just change them all? Sorry I missed these before 🤦

Reply:
I've added these as a separate commit